lib/core/bootstrap.class.ts

```diff
@@ -1,8 +1,69 @@
-import { App, IApp } from './app.class'
+import 'reflect-metadata'
+import { Server, IRouteConfiguration } from 'hapi'
+import { IApp, IAppStatic, IAppConfig, IModule, IModuleStatic, IModuleConfig, IPlugin, IPluginStatic, IPluginConfigurator, IPluginConfiguratorStatic, IRegister } from './interfaces'
+import * as _ from 'lodash'
 
-export function bootstrap(...BootstrapedApps: Array<any>): void {
+const DEFAULT_APP_CONFIG = {
+  port: 3000
+}
+
+export function bootstrap(...BootstrapedApps: Array<IAppStatic>): IApp {
+  let apps: Array<IApp> = []
   for (let BootstrapedApp of BootstrapedApps) {
-    let app: IApp = App.apps.get(BootstrapedApp.name)
-    app.start()
+    let server: Server = new Server()
+    let app: IApp = new BootstrapedApp(server)
+
+    server.connection(_.extend(Reflect.getMetadata('hapiour:config', BootstrapedApp), DEFAULT_APP_CONFIG))
+    try {app.onInit()} catch(err) {}
+    server.register(getPluginsRecurs(Reflect.getMetadata('hapiour:plugins', BootstrapedApp)), (err: any) => {
+      try {app.onRegister()} catch(err) {}
+      server.route(getRoutesWithConfigRecurs(BootstrapedApp))
+      server.start((err: any) => {
+        try {app.onStart()} catch(err) {}
+      })
+    })
+    apps.push(app)
+  }
+  return apps
+}
+
+function getPluginsRecurs(Plugins: Array<IPluginStatic|IPlugin|IPluginConfiguratorStatic|Array<IPluginStatic|IPlugin|IPluginConfiguratorStatic>>): Array<IPlugin> {
+  let plugins: Array<IPlugin> = []
+  for (let Plugin of Plugins) {
```

Comment:
can't find a way to properly use it with typescript ... got an error about lodashimplicitearraywrapper...